exp/zapslog/slog.go

```diff
@@ -157,33 +213,52 @@ func (h *Handler) Handle(ctx context.Context, record slog.Record) error {
 		}
 	}
 
-	fields := make([]zapcore.Field, 0, record.NumAttrs())
+	fields := _fieldSlices.Get()
+	defer fields.Return()
+
 	record.Attrs(func(attr slog.Attr) {
-		fields = append(fields, convertAttrToField(attr))
+		converted := convertAttrToFields(attr)
+		fields.Extend(converted)
+		converted.Return()
```

Comment:
I think if you take my suggestion to make `convert` take a `*fieldSlice`,
this won't need to borrow a new slice at all.
```suggestion
		convertAttrToFields(attr, &fields)
```